builder/dockerfile/parser/parser.go

```diff
@@ -101,19 +103,24 @@ func parseLine(line string) (string, *Node, error) {
 // Parse is the main parse routine.
 // It handles an io.ReadWriteCloser and returns the root of the AST.
 func Parse(rwc io.Reader) (*Node, error) {
+	currentLine := 0
 	root := &Node{}
+	root.StartLine = currentLine
 	scanner := bufio.NewScanner(rwc)
 
 	for scanner.Scan() {
 		scannedLine := strings.TrimLeftFunc(scanner.Text(), unicode.IsSpace)
+		currentLine++
 		line, child, err := parseLine(scannedLine)
 		if err != nil {
 			return nil, err
```

Comment:
I think the return's need some extra thought to make sure that root.EndLine is set properly since setting at the of the func won't get called on these early return statements.